src/replica/bulk_load/replica_bulk_loader.cpp

```diff
@@ -590,9 +613,17 @@ void replica_bulk_loader::download_sst_file(const std::string &remote_dir,
     METRIC_VAR_INCREMENT_BY(bulk_load_download_file_bytes, f_size);
 
     // download next file
-    if (file_index + 1 < _metadata.files.size()) {
-        const file_meta &next_f_meta = _metadata.files[file_index + 1];
-        _download_files_task[next_f_meta.name] =
+    get_f_meta = true;
```

Comment:
This judgement is duplicate with the beginning of this function, right?

I think it's because we need the f_meta.name when emplace _download_files_task, I didn't find where the string key is used, is it possible to use int as the key? Then we can just remove the duplicate code.

Reply:
The `replica_bulk_loader::clear_bulk_load_states` function executes `_metadata.files.clear()` and `cleanup_download_tasks()`. 
Since `cleanup_download_tasks()` does not halt the ongoing `replica_bulk_loader::download_sst_file`, there's a need for locking access to `_metadata.files` within `download_sst_file` to prevent concurrency issues.

Reply:
So if pop a file from `_metadata.files ` to download once for `download_sst_file()`,  it would be safe, it's only needed to add lock when pop a file.
That is to say, pass a copy of `file_meta` object popped to `download_sst_file()` instead of the index.